src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/casenotes/CaseNotesSynchronisationService.kt

```diff
@@ -27,28 +29,41 @@ class CaseNotesSynchronisationService(
   }
 
   suspend fun caseNoteInserted(event: CaseNotesEvent) {
+    val telemetry = event.toTelemetryProperties()
+
     if (event.auditModuleName == "DPS_SYNCHRONISATION") {
-      telemetryClient.trackEvent("casenotes-synchronisation-skipped", event.toTelemetryProperties())
+      telemetryClient.trackEvent("casenotes-synchronisation-skipped", telemetry)
       return
     }
     val nomisCaseNote = nomisApiService.getCaseNote(event.caseNoteId)
     caseNotesService.upsertCaseNote(nomisCaseNote.toDPSCreateSyncCaseNote()).apply {
       tryToCreateCaseNoteMapping(
         event,
         this.caseNoteId!!,
-      )
+      ).also { mappingCreateResult ->
+        val mappingSuccessTelemetry =
+          (if (mappingCreateResult == MAPPING_CREATED) mapOf() else mapOf("mapping" to "initial-failure"))
+        val additionalTelemetry = mappingSuccessTelemetry + ("dpsCaseNoteId" to this.caseNoteId.toString())
+
+        telemetryClient.trackEvent(
+          "casenotes-synchronisation-created-success",
+          telemetry + additionalTelemetry,
+        )
```

Comment:
Might this be easier to follow if you just keep adding to `telemetry` when you need to instead of introducing additional variables? I had to read this a couple of times to get it.